pragma-node/src/handlers/entries/subscribe_to_entry.rs

```diff
@@ -0,0 +1,390 @@
+        self.perp_pairs.extend(pairs);
+    }
+
+    fn remove_spot_pairs(&mut self, pairs: &[String]) {
+        for pair in pairs {
+            self.spot_pairs.remove(pair);
+        }
+    }
+
+    fn remove_perp_pairs(&mut self, pairs: &[String]) {
+        for pair in pairs {
+            self.perp_pairs.remove(pair);
+        }
+    }
+
+    /// Get the subscribed spot pairs.
+    fn get_subscribed_spot_pairs(&self) -> Vec<String> {
+        self.spot_pairs.iter().cloned().collect()
+    }
+
+    /// Get the subscribed perps pairs (without suffix).
+    fn get_subscribed_perp_pairs(&self) -> Vec<String> {
+        self.perp_pairs.iter().cloned().collect()
+    }
+
+    /// Get the subscribed perps pairs with the MARK suffix.
+    fn get_fmt_subscribed_perp_pairs(&self) -> Vec<String> {
+        self.perp_pairs
+            .iter()
+            .map(|pair| format!("{}:MARK", pair))
+            .collect()
+    }
+
+    /// Get all the currently subscribed pairs.
+    /// (Spot and Perp pairs with the suffix)
+    fn get_fmt_subscribed_pairs(&self) -> Vec<String> {
+        let mut spot_pairs = self.get_subscribed_spot_pairs();
+        let perp_pairs = self.get_fmt_subscribed_perp_pairs();
+        spot_pairs.extend(perp_pairs);
+        spot_pairs
+    }
+}
+
+#[utoipa::path(
+    get,
+    path = "/node/v1/data/subscribe",
+    responses(
+        (
+            status = 200,
+            description = "Subscribe to a list of entries",
+            body = [SubscribeToEntryResponse]
+        )
+    )
+)]
+pub async fn subscribe_to_entry(
+    ws: WebSocketUpgrade,
+    State(state): State<AppState>,
+) -> impl IntoResponse {
+    ws.on_upgrade(move |socket| handle_channel(socket, state))
+}
+
+/// Interval in milliseconds that the channel will update the client with the latest prices.
+const CHANNEL_UPDATE_INTERVAL_IN_MS: u64 = 500;
+
+/// Handle the WebSocket channel.
+async fn handle_channel(mut socket: WebSocket, state: AppState) {
+    let waiting_duration = Duration::from_millis(CHANNEL_UPDATE_INTERVAL_IN_MS);
+    let mut update_interval = interval(waiting_duration);
+    let mut subscription: CurrentSubscription = Default::default();
+
+    loop {
+        tokio::select! {
+            Some(msg) = socket.recv() => {
+                if let Ok(Message::Text(text)) = msg {
+                    handle_message_received(&mut socket, &state, &mut subscription, text).await;
+                }
+            },
+            _ = update_interval.tick() => {
+                match send_median_entries(&mut socket, &state, &subscription).await {
+                    Ok(_) => {},
+                    Err(_) => break
+                };
+            }
+        }
+    }
+}
+
+/// Handle the message received from the client.
+/// Subscribe or unsubscribe to the pairs requested.
+async fn handle_message_received(
+    socket: &mut WebSocket,
+    state: &AppState,
+    subscription: &mut CurrentSubscription,
+    message: String,
```

Comment:
```suggestion
    message: Message,
```

we need to match on different message types, you can reuse what's done in other ws endpoint

Reply:
done!